osu.Game/Beatmaps/BeatmapStore.cs

```diff
@@ -34,11 +34,12 @@ public void Add(BeatmapSetInfo beatmapSet)
 
             foreach (var beatmap in beatmapSet.Beatmaps.Where(b => b.Metadata != null))
             {
+                // check local context for metadata so we can reuse duplicates from the same set

```

Comment:
i was hoping for explanation on what `.Local` and `.Set` are doing here to achieve this..